### PR TITLE
[BEAM-551] Add support for native PubSubIO

### PR DESCRIPTION
**Pull request by sammcveety**
R: @davorbonaci

### CODE REVIEW

File: sdks/java/core/src/main/java/org/apache/beam/sdk/options/ValueProvider.java [hunk 1]
```diff
@@ -134,6 +134,20 @@ public boolean isAccessible() {
       return value.isAccessible();
     }
 
+    /**
+     * Return the property name associated with this provider.
```
Comment (davorbonaci) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/options/ValueProvider.java]: Returns (for generated Javadoc style purposes)

File: sdks/java/core/src/main/java/org/apache/beam/sdk/options/ValueProvider.java [hunk 2]
```diff
@@ -134,6 +134,20 @@ public boolean isAccessible() {
...
+     */
+    public String propertyName() {
```
Comment (davorbonaci) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/options/ValueProvider.java]: test?

**Comment by sammcveety** [comment]
Done

**Comment by davorbonaci** [comment]
LGTM, modulo checkstyle.
```
[INFO] There is 1 error reported by Checkstyle 6.19 with beam/checkstyle.xml ruleset.
[ERROR] src/test/java/org/apache/beam/sdk/options/ValueProviderTest.java:[249,27] (naming) LocalVariableName: Name 'double_nvp' must match pattern '^[a-z][a-zA-Z0-9]*$'.
```

**Comment by sammcveety** [comment]
Done

**Comment by davorbonaci** [comment]
retest this please

**Pull request closed without merging**